lib/hedge_fund_interview/prisms/fetch_signals_history.ex

```diff
@@ -0,0 +1,78 @@
+defmodule HedgeFundInterview.Prisms.FetchSignalsHistory do
+  use Lux.Prism,
+    name: "Fetch Signals History",
+    description:
+      "Fetch the history of signals exchanged with a specific sender in the context of a job opening.",
+    input_schema: %{
+      type: "object",
+      properties: %{
+        sender: %{type: "string"},
+        job_opening_id: %{type: "string"}
+      }
+    }
+
+  alias HedgeFundInterview.Schemas
+
+  @interview_message_schema_id Schemas.InterviewMessageSchema.signal_schema_id()
+  @reject_message_schema_id Schemas.RejectMessageSchema.signal_schema_id()
+  @shortlist_message_schema_id Schemas.ShortlistMessageSchema.signal_schema_id()
+  @messages_count_response_schema_id Schemas.MessagesCountResponseSchema.signal_schema_id()
+  @error_schema_id Schemas.ErrorSchema.signal_schema_id()
+
+  @signal_ids_to_names %{
+    @interview_message_schema_id => "interview_message",
+    @reject_message_schema_id => "reject_message",
+    @shortlist_message_schema_id => "shortlist_message",
+    @messages_count_response_schema_id => "messages_count_response",
+    @error_schema_id => "error"
+  }
+
+  @signals_history_path "/job_positions/:job_position_id/:ans_name/raw_signals"
+  @default_history_limit 20
```

Comment:
This is not ideal. IMO we should fetch by job application ID (after we start including it in the signals) so we can retrieve a single conversation easily.